include/z64camera.h

```diff
@@ -1363,8 +1363,8 @@ typedef struct {
     /* 0x104 */ f32 playerGroundY;
     /* 0x108 */ Vec3f floorNorm;
     /* 0x114 */ f32 waterYPos;
-    /* 0x118 */ s32 waterPrevCamIdx;
-    /* 0x11C */ s32 waterPrevCamSetting;
+    /* 0x118 */ s32 waterBgCamDataIndex;
```

Comment:
this looks like the bgCamDataIndex before entering a waterbox

Reply:
It traces to 
`waterBgCamDataIndex = Camera_GetWaterBoxBgCamDataIndex(camera, &waterY)`
Which then:
`bgCamDataIndex = WaterBox_GetBgCamDataIndex(&camera->play->colCtx, waterBox);`
Which then:
```
u32 WaterBox_GetBgCamDataIndex(CollisionContext* colCtx, WaterBox* waterBox) {
    u32 prop = waterBox->properties >> 0;

    return prop & 0xFF;
}
```
So it is actually extracting a value from `waterBox->properties`, which it then later gets put into:
`Camera_ChangeBgCamDataIndex(camera, camera->prevBgCamDataIndex);`

That's the extent of my knowledge, I haven't dug deeper than that, particularly `waterBox->properties`. But it seems like it's actually extracting a value from the water box, instead of just storing a previous value?

Reply:
In `Camera_UpdateWater`, `camera->waterBgCamDataIndex` does not get its value from `waterBgCamDataIndex`

The local variable name is fine (I think), it's the `Camera` struct member that I think is the "prev" (`camera->waterBgCamDataIndex = camera->bgCamDataIndex;`)

Reply:
Ah I see, yes you're correct. `waterBgCamDataIndex -> prevBgCamDataIndex` would indeed make sense, but there's already a `prevBgCamDataIndex` in the camera struct, which seems to be used for 1) opening large chests 2) Having the camera flip around to look at Link (ocarina, get item, call epona, etc...). Hmmm

Reply:
Taking inspiration from `gLightConfigAfterUnderwater`,
`bgCamDataIndexAfterUnderwater` ?

Reply:
Recalling the discussion we had about `prev` vs. `next`, wouldn't the parallels be:
`prev` -> `before`
`next` -> `after`

So would `bgCamDataIndexBeforeUnderwater`
make more sense?